src/components/serviceRequest/ServiceRequestDialog.tsx

```diff
@@ -74,9 +147,9 @@ function ServiceRequestDialog(props: ServiceRequestDialogProps) {
     // TODO add error handling scenario
     try {
       await clientService.newRequest(
-        serviceInformationState,
-        clientInformationState,
-        petInformationState,
+        newTicket,
```

Comment:
Are they always a new ticket, client, and pet? Or is it sometimes an existing object

Reply:
Its not always new, the client or pet might be matched by email or name. Will update variable name to reflect

Reply:
Actually, that's not totally accurate. In this component, the client and pet would be just user input. The Service handles determining if the client or pet is new or existing. This component doesn't know which.

We plan to change this UI in the future so it doesn't do an implicit match by email/name to an existing record and instead makes it more clear that you are either selecting an existing client/pet or creating a new one.

I only carried forward the previous implicit matching behavior. I didn't question it at the time :)

Reply:
I am confused, I thought the` newticket `refers to information that is being entered by the user and we use the `ticketid` flag to determine if its a `newticket` or a saved one? But since this call is targeting the `clientService.newRequest` won't it always be a` newTicket` being passed here?

Reply:
@tristan-88 where is the being checked if the ticket is a new one or a saved one?